web-client/public/js/upload.js

```diff
@@ -301,30 +329,104 @@ export const upload = function (sliderObject, sliderGroupController, drawGraph,
+    $("#edge-weight-normalization-factor-menu").on("change", function () {
+        synchronizeNormalizationValues($("#edge-weight-normalization-factor-menu").val());
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#dashedGrayLineButton").on("change", function () {
-    // displayNetwork(currentNetwork, name, normalization, grayThreshold);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    var GREY_EDGE_THRESHOLD_MENU = "#gray-edge-threshold-menu";
+    var GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR = "#grayThresholdInput";
+    var GREY_EDGE_THRESHOLD_TEXT_SIDEBAR = "#grayThresholdValue";
+    // var DEFAULT_GRAY_EDGE_THRESHOLD = 5;
+
+    // Gray Edge Controller
+
+    var grayEdgeInputValidator = function (value) {
+        if (value < 0) {
+            return 0;
+        } else if (value > 100) {
+            return 100;
+        } else {
+            return value;
+        }
+    };
+
+    var updateGrayEdgeValues = function (value) {
+        var validatedInput = grayEdgeInputValidator(value);
+        $(GREY_EDGE_THRESHOLD_TEXT_SIDEBAR).text(validatedInput + "%");
+        $(GREY_EDGE_THRESHOLD_MENU).val(validatedInput);
+        $(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).val(validatedInput / 100);
+    };
+
+    $(GREY_EDGE_THRESHOLD_MENU).on("change", function () {
+        var value = Math.round(($(GREY_EDGE_THRESHOLD_MENU).val()));
+        updateGrayEdgeValues(value);
+        drawGraph(currentNetwork, sliders, nodeColoring);
+    });
+
+    $(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).on("change", function () {
+        var value = Math.round(($(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).val() * 100));
+        updateGrayEdgeValues(value);
+        drawGraph(currentNetwork, sliders, nodeColoring);
+    });
+
+    // Dashed Gray Line Controller
+
+    var GREY_EDGES_DASHED_MENU = "#grey-edges-dashed-menu";
+    var GREY_EDGES_DASHED_SIDEBAR = "#dashedGrayLineButton";
+
+    var syncGreyEdgesAsDashedOptions = function (showAsDashed) {
+        if (showAsDashed) {
+            $(GREY_EDGES_DASHED_MENU + " span").addClass("glyphicon-ok");
+            $(GREY_EDGES_DASHED_MENU).prop("checked", "checked");
+            $(GREY_EDGES_DASHED_SIDEBAR).prop("checked", "checked");
+        } else {
+            $(GREY_EDGES_DASHED_MENU + " span").removeClass("glyphicon-ok");
+            $(GREY_EDGES_DASHED_MENU).removeProp("checked");
+            $(GREY_EDGES_DASHED_SIDEBAR).removeProp("checked");
+        }
+        drawGraph(currentNetwork, sliders, nodeColoring);
+    };
+
+    $(GREY_EDGES_DASHED_MENU).click(function () {
+        $(GREY_EDGES_DASHED_MENU).prop("checked") ?
+            syncGreyEdgesAsDashedOptions(false) : syncGreyEdgesAsDashedOptions(true);
+    });
+
+    $(GREY_EDGES_DASHED_SIDEBAR).on("change", function () {
+        $(GREY_EDGES_DASHED_SIDEBAR).prop("checked") ?
+            syncGreyEdgesAsDashedOptions(true) : syncGreyEdgesAsDashedOptions(false);
```

Comment:
.....especially when you change these functions to just say:
```javascript
syncGrayEdgesAsDashedOptions($(GREY_EDGES_DASHED_MENU.prop("checked"));
/* replace with GREY_EDGES_DASHED_SIDEBAR for the other event handler */
```

Reply:
👍